### PR TITLE
fix compile issue using old 2019 msvc compiler.

### PR DESCRIPTION
**Pull request by bluesky013**
Re: #

If the constructor of the derived class declares noexcept, but the corresponding constructor of the parent class does not，the compiler will delete the constructor of the parent class.

_MSC_VER : 1924

### Changelog

* fix compile issue using old 2019 msvc compiler.

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/renderer/pipeline/custom/details/SerializationUtils.h [hunk 1]
```diff
@@ -62,13 +62,21 @@ void save(OutputArchive& ar, const T& v) {
 
 template <class T,
           std::enable_if_t<
-              std::is_enum_v<T> || std::is_integral_v<T> || std::is_floating_point_v<T>,
+              std::is_integral_v<T> || std::is_floating_point_v<T>,
               bool> = false>
 void load(InputArchive& ar, T& v) {
-    static_assert(!std::is_enum_v<T> || sizeof(T) <= sizeof(uint32_t)); // enum can only be 1, 2, 4 bytes
+    static_assert(sizeof(T) <= sizeof(uint32_t)); // enum can only be 1, 2, 4 bytes
```
Comment (minggo) [inline native/cocos/renderer/pipeline/custom/details/SerializationUtils.h:68]: Change ``<= to `==`?

File: native/cocos/renderer/pipeline/custom/details/SerializationUtils.h [hunk 2]
```diff
@@ -62,13 +62,21 @@ void save(OutputArchive& ar, const T& v) {
...
     v = static_cast<T>(ar.readNumber());
 }
 
+template <class T,
+          std::enable_if_t<std::is_enum_v<T>,
+              bool> = false>
+void load(InputArchive& ar, T& v) {
+    v = static_cast<T>(static_cast<typename std::underlying_type<T>::type>(ar.readNumber()));
+}
```
Comment (minggo) [inline native/cocos/renderer/pipeline/custom/details/SerializationUtils.h:78]: Why need special operation for enum?

File: native/cocos/renderer/pipeline/custom/RenderCommonTypes.h [hunk 3]
```diff
@@ -278,7 +278,7 @@ struct LightInfo {
     uint32_t level{0};
 };
 
-enum class DescriptorTypeOrder {
+enum class DescriptorTypeOrder : uint32_t {
```
Comment (minggo) [inline native/cocos/renderer/pipeline/custom/RenderCommonTypes.h]: Why need changing to `uint32_t`? Does all enum type change to `uint32_t` now?

File: native/cocos/renderer/pipeline/custom/details/SerializationUtils.h (hunk 1 above)
Comment (bluesky013) [reply native/cocos/renderer/pipeline/custom/details/SerializationUtils.h:68]: This specialization supports `bool`, `char`, `short`, `int`, unsigned included.

File: native/cocos/renderer/pipeline/custom/details/SerializationUtils.h (hunk 2 above)
Comment (bluesky013) [reply native/cocos/renderer/pipeline/custom/details/SerializationUtils.h:78]: Cast from double to enum might not be supported.

File: native/cocos/renderer/pipeline/custom/RenderCommonTypes.h (hunk 3 above)
Comment (bluesky013) [reply native/cocos/renderer/pipeline/custom/RenderCommonTypes.h]: This modification might not be necessary.
`load` specialization uses underlying_type, and default underlying type of enum class is int. Not all enum type need to change.

Comment (minggo) [reply native/cocos/renderer/pipeline/custom/RenderCommonTypes.h]: What i think is that, we should have a rule for enum. If it requires special type, then should add some comments.

File: native/cocos/renderer/pipeline/custom/details/SerializationUtils.h (hunk 2 above)
Comment (minggo) [reply native/cocos/renderer/pipeline/custom/details/SerializationUtils.h:78]: K, then please add some comment.

Comment (minggo) [reply native/cocos/renderer/pipeline/custom/details/SerializationUtils.h:78]: K, then please add some comment.

**Review by minggo — APPROVED** [review]

**Comment by bluesky013** [comment]
@cocos-robot run test cases

**Review by star-e — APPROVED** [review]

**Pull request merged**